plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -490,4 +518,66 @@ protected Session joinPushdownEnabled(Session session)
                 .setCatalogSessionProperty(session.getCatalog().orElseThrow(), "join_pushdown_enabled", "true")
                 .build();
     }
+
+    @Test(timeOut = 60_000)
+    public void testCancellation()
+            throws Throwable
+    {
+        if (!hasBehavior(SUPPORTS_CANCELLATION)) {
+            throw new SkipException("Cancellation is not supported by given connector");
+        }
+
+        Duration minimalQueryDuration = new Duration(1, MINUTES);
+
+        // verify that long lasting query is correct
+        assertQuerySucceeds("EXPLAIN " + getLongLastingQuery(minimalQueryDuration));
+
+        Set<RemoteDatabaseEvent> remoteEvents = captureRemoteEventsDuring(() -> {
+            String longLastingQuery = getLongLastingQuery(minimalQueryDuration);
+            Future<?> future = threadPool.submit(() -> assertQueryFails(longLastingQuery, "Query killed. Message: Killed by test"));
+            QueryId queryId = getQueryId(longLastingQuery);
+            sleep(1_000); // to make sure remote data source is queried
```

Comment:
You don't need to assume how much it takes, since `getRemoteDatabaseEvents` allows you to check that the remote query has been started.

Reply:
`getRemoteDatabaseEvents` may have other queries that has started during that time. It is not an easy task to recognized what remote query was generated for given Trino query.

Reply:
Correct, that's another reason to  embed some string in these queries to be able to tell them apart.
Otherwise, the
```
        assertThat(remoteEvents.stream())
                .map(RemoteDatabaseEvent::getStatus)
                .contains(CANCELLED);
```

means  "Something got cancelled"  instead of "our query got cancelled".